src/assignmentOverride.test.js

```diff
@@ -76,39 +76,35 @@ describe('AssignmentOverride', () => {
     it('creates an assignment on the test track server', () => {
       testContext.override.persistAssignment();
 
-      expect($.ajax).toHaveBeenCalledTimes(1);
-      expect($.ajax).toHaveBeenCalledWith('http://testtrack.dev/api/v1/assignment_override', {
-        method: 'POST',
-        dataType: 'json',
-        crossDomain: true,
-        headers: {
-          Authorization: 'Basic dGhlX3VzZXJuYW1lOnRoZV9wYXNzd29yZA==' // Base64 of 'the_username:the_password'
-        },
-        data: {
+      expect(client.post).toHaveBeenCalledTimes(1);
+      expect(client.post).toHaveBeenCalledWith(
+        'http://testtrack.dev/api/v1/assignment_override',
+        {
           visitor_id: 'visitorId',
           split_name: 'jabba',
           variant: 'cgi',
           context: 'spec',
           mixpanel_result: 'success'
+        },
+        {
+          crossDomain: true,
+          headers: {
+            Authorization: 'Basic dGhlX3VzZXJuYW1lOnRoZV9wYXNzd29yZA==' // Base64 of 'the_username:the_password'
+          }
         }
-      });
+      );
     });
 
     it('logs an error if the request fails', () => {
-      $.ajax = jest.fn().mockImplementation(function() {
-        return $.Deferred().rejectWith(null, [
-          { status: 500, responseText: 'Internal Server Error' },
-          'textStatus',
-          'errorThrown'
-        ]);
+      client.post = jest.fn().mockRejectedValue({ status: 500, statusText: 'Internal Server Error' });
+
+      // eslint-disable-next-line jest/valid-expect-in-promise
+      testContext.override.persistAssignment().then(() => {
```

Comment:
Same comment here about returning the Promise:

```suggestion
      return testContext.override.persistAssignment().then(() => {
```